### PR TITLE
Feat: split variables and get rid of releases

### PR DESCRIPTION
**Pull request by erikzwart**
Changed datamodel (molgenis.csv) #8669-4:
VarriableValues -> [SourceVariableValues, TargetVariableValues]
Variables -> [SourceVariables, TargetVariables]
RepeatedVariables -> [RepeatedSourceVariables, RepeatedTargetVariables]

### CODE REVIEW

**Comment by mswertz** [comment]
I am reviewing. I will add observations here.

Observations:
* models are not one-on-one with Networks. Models should exist independent of networks. I.e. need their own name. I would recommend we try to again make SourceTables/TargetTables and SourceVariables/TargetVariables more similar by having both SourceDataDictionaries and TargetDataDictionaries

Slightly less important observations that may take more discussion but are otherwise only window dressing:
* We should think about naming. It is confusing that we use 'resource' and 'source', and confusing that we use 'model' and 'target'. We might reconside to 'source-' and 'cdm-' as prefixes? And thus call models 'CommonDataModels'.

**Comment by mswertz** [comment]
[#8669-4 releases eruit en variabele model splitsen in collected vs harmonized](https://trello.com/c/sx5RpTjo/328-8669-4-releases-eruit-en-variabele-model-splitsen-in-collected-vs-harmonized)

**Comment by mswertz** [comment]
I have updated UX, please review.

**Review by BrendaHijmans — CHANGES_REQUESTED** [review]
-import error messages are not functioning. I am getting "Import failed: Transaction failed: Cannot commit transaction in .....ms." when data is not valid and no specifics are shown. Specifics are shown in other versions of the software for the same data, so this seems to be specific for this PR.
-variable explorer is showing repeated variables separately, this was not the case before. The display of the repeats does not include a variable name (shoudl be taken form the first repeat, which is defined in TargetVariables).
-Cohort view is not showing: "Error: Validation error of type FieldUndefined: Field 'SourceTables_agg' in type 'Query' is undefined @ 'SourceTables_agg'"

File: apps/catalogue/src/store/actions.js [hunk 1]
```diff
@@ -287,6 +281,9 @@ export default {
     const resp = await request("graphql", query, { filter }).catch((e) =>
       console.error(e)
     );
+
+    console.log(resp);
```
Comment (connoratrug) [inline apps/catalogue/src/store/actions.js:285]: remove debug code

**Review by connoratrug — CHANGES_REQUESTED** [review]
Errors of viewing variable detail

```variableRepository.js:24 fetch variableDetails details failed: Error: Error: Validation error of type FieldUndefined: Field 'fromVariablesInOtherTables' in type 'VariableMappings' is undefined @ 'TargetVariables/mappings/fromVariablesInOtherTables': {"response":{"errors":[{"message":"Error: Validation error of type FieldUndefined: Field 'fromVariablesInOtherTables' in type 'VariableMappings' is undefined @ 'TargetVariables/mappings/fromVariablesInOtherTables'"}],"status":400,"headers":{"map":{"connection":"keep-alive","content-type":"application/json","date":"Mon, 10 Jan 2022 14:39:16 GMT","server":"nginx/1.19.2","transfer-encoding":"chunked"}}},"request":{"query":"query TargetVariables($filter: TargetVariablesFilter) {\n  TargetVariables(limit: 1, filter: $filter) {\n    name\n    label\n    description\n    format {\n      name\n    }\n    unit {\n      name\n    }\n    permittedValues {\n      value\n      label\n      order\n      isMissing\n      ontologyTermIRI\n    }\n    mappings {\n      syntax\n      description\n      match {\n        name\n      }\n      fromDataDictionary {\n        resource {\n          pid\n        }\n      }\n      fromVariable {\n        name\n      }\n      fromVariablesInOtherTables {\n        table {\n          name\n        }\n        name\n      }\n      fromTable {\n        dataDictionary {\n          resource {\n            pid\n          }\n          version\n        }\n        name\n      }\n    }\n    repeats {\n      name\n      mappings {\n        syntax\n        description\n        match {\n          name\n        }\n        fromDataDictionary {\n          resource {\n            pid\n          }\n        }\n        fromVariable {\n          name\n        }\n        fromTable {\n          dataDictionary {\n            resource {\n              pid\n            }\n            version\n          }\n          name\n        }\n      }\n    }\n  }\n}\n","variables":{"filter":{"name":{"equals":"alc"},"dataDictionary":{"equals":[{"resource":{"pid":"LongITools_CDM"},"version":"1.0.0"}]}}}}} ```

**Comment by mswertz** [comment]
there is also a platform bug because RepeatSourceVariables is fairly long, within queries we go beyond 64 characters. So I believe we need to change SqlQuery class to use short identifiers instead of a meaningful concatenation of table names. Might be tricky :-)

**Comment by mswertz** [comment]
> -Cohort view is not showing: "Error: Validation error of type FieldUndefined: Field 'SourceTables_agg' in type 'Query' is undefined @ 'SourceTables_agg'"

Did you use correct data model as in the pull request?

**Comment by mswertz** [comment]
I have updated the pull request to address most issues, can you retest?

**Review by connoratrug — CHANGES_REQUESTED** [review]
- view variables 
- see list of vars ( lost of lifecycle in example )
- select lifecycle in filter
- expected: only lifecycle vars
- actual: empty variable list

**Comment by mswertz** [comment]
> -import error messages are not functioning. I am getting "Import failed: Transaction failed: Cannot commit transaction in .....ms." when data is not valid and no specifics are shown. Specifics are shown in other versions of the software for the same data, so this seems to be specific for this PR.

This is a bug that we need to fix

**Comment by mswertz** [comment]
To fix the repeated variables that are now also shown in the variable-explorer, we will change the data model such that RepeatedTargetVariables are not in the TargetVariables. Therefore create a common superclass table 'AllTargetVariabes' (and for symmetry also AllSourceVariables).

**Review by BrendaHijmans — APPROVED** [review]
variable explorer works after datamodel update.
i found no other issues specific to this PR

**Review by connoratrug — COMMENTED** [review]
- Go to variable explorer
- view harmonisation grid ( some are green some are not)
- Add filer from tree ( or list  ) that results in empty list 
- Expected; empty list
- Result: spinner

**Review by connoratrug — COMMENTED** [review]
Padding seems to be missing / inconsistent, suggest to use bs defaults
 
Screenshot 2022-01-12 at 15 46 45

**Review by connoratrug — COMMENTED** [review]
Tree does not take up available space below block, this makes is hard to browse the crapped tree

Screenshot 2022-01-12 at 15 46 53

**Review by connoratrug — COMMENTED** [review]
strange :hover styling

Screenshot 2022-01-12 at 15 49 02

**Review by connoratrug — APPROVED** [review]

**Comment by connoratrug** [comment]
approved; see notes on empty lists and styling

**Comment by mswertz** [comment]
Good points from Connor on the layout. Added a story to the backlog to address these.

Meanwhile: bugfix for long table names #871 and bugfix for missing import error message #872 have been made

**Pull request merged**